python2/koans/about_methods.py

```diff
@@ -74,13 +74,13 @@ def test_functions_without_self_arg_are_global_functions(self):
         def function_with_the_same_name(a, b):
             return a * b
 
-        self.assertEqual(__, function_with_the_same_name(3, 4))
+        self.assertEqual(12, function_with_the_same_name(3, 4))
 
     def test_calling_methods_in_same_class_with_explicit_receiver(self):
         def function_with_the_same_name(a, b):
```

Comment:
Did you notice that this function was declared within another function?

Just like everything else in Python, function definitions are actually create an objects with the name of the functions, so it makes sense that you can define them pretty much anywhere. A function is an object that executes the code in the function body when called with parameters.

Notice that the function definition does not execute the function, just creates the function object.

Even weirder, Class definitions are executable objects in Python too, but objects that return an instance of themselves when used. This is in stark contrast to C++, where Classes are types, and it's the reason why you can pass classes (not instances, classes) as Python parameters, for instance.

Ir remark this because this because understanding this is at the core of understanding Python, and will help you a lot to understand why OOP works the way it does in Python.